lib/std/coff.zig

```diff
@@ -1214,6 +1214,11 @@ pub const Coff = struct {
         return Strtab{ .buffer = self.data[offset..][0..size] };
     }
 
+    pub fn strtabRequired(self: *const Coff) bool {
```

Comment:
```
                // The string table is not mapped into memory by the loader, so if a section name is in the
                // string table then we have to map the full image file from disk. This can happen when
                // a binary is produced with -gdwarf, since the section names are longer than 8 bytes.
                if (coff_obj.strtabRequired()) {
```

If we were always loading the COFF from disk, it would be redundant, however in most cases we can use the COFF image that was memory mapped by the loader into the address space, without needing to hit the disk - since all we need from it is to get the PDB path which we'll then load from disk.

The exception is if `-gdwarf` was used, which means:
- The COFF has section names longer than 8 bytes, triggering use of the string table (which isn't mapped by the loader)
- We need the debug sections from the COFF to get source line info

In that case, I memory map the full COFF file from disk, and load the sections from there.

Reply:
I am confused - how come we don't need to do the same check for ELF/MachO? Is COFF not mmaped by default?

Reply:
Since my changes in https://github.com/ziglang/zig/pull/14247 (which actually regressed `-gdwarf` support, which this now fixes), we don't mmap the COFF image ourselves, instead reading it directly from what the loader has mapped.

The problem was that (and I didn't realize this at the time), is the entire COFF image isn't mapped - some sections are zeroed out, including the string table.

This change fixes that regression by mapping a view of the COFF file on disk into memory, and using that to load the string table, and `.debug_*` sections.

Reply:
Ahhh, I see, thanks for reminding me!